src/execution_plan/ops/op_join.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018-2019 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#include "op_join.h"
+
+/* Forward declarations. */
+static Record JoinConsume(OpBase *opBase);
+static OpResult JoinReset(OpBase *opBase);
+static OpResult JoinInit(OpBase *opBase);
+
+OpBase *NewJoinOp(const ExecutionPlan *plan) {
+	OpJoin *op = malloc(sizeof(OpJoin));
+	op->stream = NULL;
+
+	// Set our Op operations
+	OpBase_Init((OpBase *)op, OPType_JOIN, "Join", JoinInit, JoinConsume, NULL, NULL, NULL, plan);
+
+	return (OpBase *)op;
+}
+
+static OpResult JoinInit(OpBase *opBase) {
+	OpJoin *op = (OpJoin *)opBase;
+	// Start pulling from first stream.
+	op->streamIdx = 0;
+	op->stream = op->op.children[op->streamIdx];
+	return OP_OK;
+}
+
+static Record JoinConsume(OpBase *opBase) {
+	OpJoin *op = (OpJoin *)opBase;
+	Record r = NULL;
+
+	while(true) {
+		// Try pulling from current stream.
+		r = OpBase_Consume(op->stream);
+		// Managed to get a record, break out.
+		if(r) break;
```

Comment:
`if(r) return r;` 
I think it is cleaner.

Reply:
Restructured.